File: packages/astro-polyglot/tests/fixtures/csharp/MathUtils.cs
/// <summary>
/// A utility class providing mathematical operations.
/// This class demonstrates .NET XML documentation conventions.
/// </summary>
public static class MathUtils
{
    /// <summary>
    /// Adds two integers and returns the result.
    /// </summary>
    /// <param name="a">The first operand.</param>
    /// <param name="b">The second operand.</param>
    /// <returns>The sum of a and b.</returns>
    public static int Add(int a, int b) => a + b;

    /// <summary>
    /// Divides one double by another.
    /// </summary>
    /// <param name="numerator">The value to be divided.</param>
    /// <param name="denominator">The value to divide by.</param>
    /// <returns>The quotient of numerator divided by denominator.</returns>
    /// <exception cref="System.DivideByZeroException">Thrown when denominator is zero.</exception>
    public static double Divide(double numerator, double denominator)
    {
        if (denominator == 0.0)
            throw new System.DivideByZeroException("Denominator must not be zero");
        return numerator / denominator;
    }

    /// <summary>
    /// Represents a 2D point with X and Y coordinates.
    /// </summary>
    public struct Point
    {
        /// <summary>The X-coordinate.</summary>
        public double X { get; set; }
        /// <summary>The Y-coordinate.</summary>
        public double Y { get; set; }

        /// <summary>
        /// Initializes a new Point at the given coordinates.
        /// </summary>
        /// <param name="x">The X-coordinate.</param>
        /// <param name="y">The Y-coordinate.</param>
        public Point(double x, double y)
        {
            X = x;
            Y = y;
        }

        /// <summary>
        /// Computes the Euclidean distance to another Point.
        /// </summary>
        /// <param name="other">The other point.</param>
        /// <returns>The distance between this point and the other point.</returns>
        public double DistanceTo(Point other)
        {
            double dx = this.X - other.X;
            double dy = this.Y - other.Y;
            return System.Math.Sqrt(dx * dx + dy * dy);
        }
    }

    /// <summary>
    /// Returns the larger of two integers.
    /// </summary>
    /// <param name="a">The first integer.</param>
    /// <param name="b">The second integer.</param>
    /// <returns>The maximum of a and b.</returns>
    public static int Max(int a, int b) => a >= b ? a : b;
}

File: packages/astro-polyglot/tests/fixtures/php/MathUtils.php
<?php
/**
 * A utility class providing mathematical operations.
 *
 * This file demonstrates PHPDoc conventions with @param, @return, @throws,
 * and @see tags.
 */
class MathUtils
{
    /**
     * Adds two integers and returns the result.
     *
     * @param int $a The first operand.
     * @param int $b The second operand.
     *
     * @return int The sum of a and b.
     */
    public static function add(int $a, int $b): int
    {
        return $a + $b;
    }

    /**
     * Divides one float by another.
     *
     * @param float $numerator   The value to be divided.
     * @param float $denominator The value to divide by.
     *
     * @return float The quotient of numerator divided by denominator.
     *
     * @throws InvalidArgumentException If denominator is zero.
     */
    public static function divide(float $numerator, float $denominator): float
    {
        if ($denominator === 0.0) {
            throw new InvalidArgumentException('Denominator must not be zero');
        }
        return $numerator / $denominator;
    }

    /**
     * Returns the larger of two integers.
     *
     * @param int $a The first integer.
     * @param int $b The second integer.
     *
     * @return int The maximum of a and b.
     */
    public static function max(int $a, int $b): int
    {
        return $a >= $b ? $a : $b;
    }
}

/**
 * Represents a 2D point with x and y coordinates.
 */
class Point
{
    /** @var float The x-coordinate. */
    public float $x;
    /** @var float The y-coordinate. */
    public float $y;

    /**
     * Creates a new Point at the given coordinates.
     *
     * @param float $x The x-coordinate.
     * @param float $y The y-coordinate.
     */
    public function __construct(float $x, float $y)
    {
        $this->x = $x;
        $this->y = $y;
    }

    /**
     * Computes the Euclidean distance to another Point.
     *
     * @param Point $other The other point.
     *
     * @return float The distance between this point and the other point.
     */
    public function distanceTo(Point $other): float
    {
        $dx = $this->x - $other->x;
        $dy = $this->y - $other->y;
        return sqrt($dx * $dx + $dy * $dy);
    }
}